spring-boot-actuator/src/main/java/org/springframework/boot/actuate/info/InfoProvider.java

```diff
@@ -0,0 +1,17 @@
+package org.springframework.boot.actuate.info;
+
+import java.util.Map;
+
+/**
+ * information provider for the info endpoint
+ *
+ * @author Meang Akira Tanaka
+ */
+public interface InfoProvider {
+
+	/**
+	 * @return a collection of information
+	 */
+	Map<String, Object> provide();
```

Comment:
Ok i will do that

Have a nice day :-)

On Wed, Jul 15, 2015, 10:25 Stéphane Nicoll <notifications@github.com>
wrote:

> In
> spring-boot-actuator/src/main/java/org/springframework/boot/actuate/info/InfoProvider.java
> <https://github.com/spring-projects/spring-boot/pull/3492#discussion_r34656295>
> :
>
> > @@ -0,0 +1,17 @@
> > +package org.springframework.boot.actuate.info;
> > +
> > +import java.util.Map;
> > +
> > +/**
> > + * information provider for the info endpoint
> > + *
> > + * @author Meang Akira Tanaka
> > + */
> > +public interface InfoProvider {
> > +
> > +	/**
> > +	 * @return a collection of information
> > +	 */
> > +	Map<String, Object> provide();
>
> If we accept this PR, we'll have to break InfoEndpoint, yes.
>
> I also think that the business of reading info.* should move to a default
> implementation of InfoProvider. You may want to review the health support
> as it has quite a lot of similarities with what you're trying to do. Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spring-projects/spring-boot/pull/3492/files#r34656295>
> .
>